day23/stringsRearrangement.py

```diff
@@ -0,0 +1,37 @@
+from itertools import permutations
+
+def stringsRearrangement(inputArray):
+    for perm in permutations(inputArray):
+        if neighborsDifferByOne(perm):
+            return True 
+    return False
+
+def neighborsDifferByOne(inputArray):
+    if len(inputArray) <= 1:
+        return True
+    return differByOne(inputArray[0], inputArray[1]) and neighborsDifferByOne(inputArray[2:])
```

Comment:
for a list `[a, b, c, d, e, f]` i think this only checks `[a, b], [c, d], [e, f]`, and skips `[b, c], [d, e]` which i think you also need to check